test/modules/message-spec.js

```diff
@@ -250,10 +270,44 @@ describe('Message class test', () => {
       assert.deepEqual(prefix, notEmptyPrefix);
     });
 
+    it('should return data according to settings [settings] [log caller prefix] [disabled]', () => {
+      const message = new Message();
+      const prefix = message.getPrefix({ LOG_CALLER_PREFIX: false });
+      assert(!{}.hasOwnProperty.call(prefix, 'module'));
+      assert(!{}.hasOwnProperty.call(prefix, 'function'));
+      assert(!{}.hasOwnProperty.call(prefix, 'project'));
+    });
+
+    it('should return data according to settings [settings] [log caller prefix] [enabled]', () => {
+      const message = new Message();
+      const prefix = message.getPrefix({ LOG_CALLER_PREFIX: true });
+      assert({}.hasOwnProperty.call(prefix, 'module'));
+      assert({}.hasOwnProperty.call(prefix, 'function'));
+      assert({}.hasOwnProperty.call(prefix, 'project'));
+    });
+
+    it('should use env prior to settings [env and settings] [log caller prefix] [enabled]', () => {
+      const message = new Message();
+      process.env.LOG_CALLER_PREFIX = 'true';
+      const prefix = message.getPrefix({ LOG_CALLER_PREFIX: false });
+      assert({}.hasOwnProperty.call(prefix, 'module'));
+      assert({}.hasOwnProperty.call(prefix, 'function'));
+      assert({}.hasOwnProperty.call(prefix, 'project'));
+    });
+
+    it('should use env prior to settings [env and settings] [log caller prefix] [disabled]', () => {
+      const message = new Message();
+      process.env.LOG_CALLER_PREFIX = 'false';
+      const prefix = message.getPrefix({ LOG_CALLER_PREFIX: true });
+      assert(!{}.hasOwnProperty.call(prefix, 'module'));
```

Comment:
probably overcomplicated your question :). I am using `{}.hasOwnProperty.call` instead of `Object.hasOwnProperty` for eslint.